packages/eslint-plugin/src/util/misc.ts

```diff
@@ -242,20 +242,25 @@ type NodeWithKey =
 function getStaticMemberAccessValue(
   node: NodeWithKey,
   { sourceCode }: RuleContext<string, unknown[]>,
-): string | undefined {
+): string | symbol | undefined {
   const key =
     node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
   if (!node.computed) {
     return key.type === AST_NODE_TYPES.Literal
       ? `${key.value}`
       : (key as TSESTree.Identifier | TSESTree.PrivateIdentifier).name;
```

Comment:
This type assertion isn't necessary if the `key` is checked within the condition, i.e.
```suggestion
  if (!node.computed) {
    const key =
      node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
    return key.type === AST_NODE_TYPES.Literal
      ? `${key.value}`
      : key.name;
```

Taking this suggestion literally will incur duplication, though, since `key` is used in the else branch too. Is there a way to make this nice?

Granted another type assertion in this very function was part of what burned us on this bug, it would be nice to avoid this one too.

Reply:
Totally agree! Refactored ✅